drivers/SmartThings/philips-hue/src/init.lua

```diff
@@ -891,65 +836,35 @@ local stray_bulb_tx, stray_bulb_rx = cosock.channel.new()
 stray_bulb_rx:settimeout(30)
 
 cosock.spawn(function()
-  local logged_stray_init = utils.log_func_wrapper(_initialize, "StrayBulb__initialize")
+  local stray_lights = {}
+  local stray_dni_to_rid = {}
+  local found_bridges = {}
+  local thread_local_driver = nil
+
   local process_strays = utils.log_func_wrapper(
-    function(driver, api_instance, strays, bridge_device_uuid)
+    function(driver, strays, bridge_device_uuid)
       local dnis_to_remove = {}
 
-      log.info_with({hub_logs = true}, "Querying bridge for devices from stray light handler")
-      Discovery.search_bridge_for_supported_devices(driver, api_instance, function(hue_driver, svc_info, device_data)
-        if not (svc_info.rid and svc_info.rtype and svc_info.rtype == "light") then return end
-
-        for light_dni, light_device in pairs(strays) do
-          local matching_v1_id = light_device.data and light_device.data.bulbId and
-              light_device.data.bulbId == device_data.id_v1:gsub("/lights/", "")
-          local matching_uuid = light_device.device_network_id == svc_info.rid or
-              light_device.device_network_id == svc_info.rid
-
-          if matching_v1_id or matching_uuid then
-            local api_key_extracted = api_instance.headers["hue-application-key"]
-            log.info_with({ hub_logs = true }, " ", light_device.label, ", re-adding")
-            log.info_with({ hub_logs = true }, string.format(
-              'Found Bridge for stray light %s, retrying onboarding flow.\n' ..
-              '\tMatching v1 id? %s\n' ..
-              '\tMatching uuid? %s\n' ..
-              '\tlight_device DNI: %s\n' ..
-              '\tlight_device Parent Assigned Key: %s\n' ..
-              '\tlight_device parent device id: %s\n' ..
-              '\tProvided bridge_device_id: %s\n' ..
-              '\tCached API key for given bridge_device_id: %s\n' ..
-              '\tCached bridge device for given API key: %s\n'
-              ,
-              light_device.label,
-              matching_v1_id,
-              matching_uuid,
-              light_device.device_network_id,
-              light_device.parent_assigned_child_key,
-              light_device.parent_device_id,
-              bridge_device_uuid,
-              Discovery.api_keys[driver:get_device_info(bridge_device_uuid).device_network_id],
-              driver.api_key_to_bridge_id[api_key_extracted]
-            ))
-            table.insert(dnis_to_remove, light_dni)
-            logged_stray_init(hue_driver, light_device, nil, nil, bridge_device_uuid)
-          end
+      for light_dni, light_device in pairs(strays) do
```

Comment:
I'm wondering if we want to keep a list of "found" strays and log any that don't find a match by the end of the for loop.  Just because I don't think we would have great visibility currently if some device never finds a match.

Reply:
:+1:

We do technically have that list, since we only remove keys from the strays list if we find a match, and that's probably a better place to print them than at the top.